### PR TITLE
Add new recipe for absent

### PR DESCRIPTION
**Pull request by rvarago**
Add recipe for version 0.0.1 of https://github.com/rvarago/absent.

Specify library name and version:  **absent/0.0.1**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/absent/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+class AbsentConan(ConanFile):
+    name = "absent"
+    description = "A simple library to compose nullable types in a generic, type-safe, and declarative style for C++"
+    homepage = "https://github.com/rvarago/absent"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "MIT"
+    author = "Rafael Varago (rvarago)"
+    topics = ("nullable-types", "composition", "monadic-interface", "declarative-programming")
+
```
Comment (uilianries) [inline recipes/absent/all/conanfile.py]:
```suggestion
    no_copy_source = True
```

**Comment by uilianries** [comment]
Olá Rafael!

Seems like you are the author of this project! That's really nice!

The CI job failed due a hook validation, please, accept my change including [no_copy_source](https://docs.conan.io/en/latest/reference/conanfile/attributes.html#no-copy-source)

Vlw Br!

**Comment by rvarago** [comment]
> Olá Rafael!
> 
> Seems like you are the author of this project! That's really nice!
> 
> The CI job failed due a hook validation, please, accept my change including [no_copy_source](https://docs.conan.io/en/latest/reference/conanfile/attributes.html#no-copy-source)
> 
> Vlw Br!

Salve! 

Thanks for the suggestion and quick response :).

I have accepted your change, thanks. 

About the CI, it happened because absent and its test_package depend on C++ 17 and so it's not possible to compile with gcc5. Would you happen to have a suggestion on how I should move forward, please? 

Vlw.

**Comment by uilianries** [comment]
@rvarago https://github.com/rvarago/conan-center-index/pull/1

**Comment by rvarago** [comment]
> @rvarago [rvarago#1](https://github.com/rvarago/conan-center-index/pull/1)


Thanks :)

**Comment by uilianries** [comment]
Well, seems like apple-clang 9 is not able to support c++17. I will need to it in that list. Only apple-clang 10 will be supported.

**Comment by danimtb** [comment]
Seems this is a real issue in the recipe. Please check the above logs in order to move the PR forward. Thanks and sorry for the noise

**Comment by uilianries** [comment]
@rvarago please, take a look at https://github.com/rvarago/conan-center-index/pull/2

I've provide a fix to the test_package.

**Comment by rvarago** [comment]
Thanks! I've applied your patch.

File: recipes/absent/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["BUILD_TESTS"] = "OFF"
+        cmake.configure(source_folder=self._source_subfolder,
+                        build_folder=self._build_subfolder)
+        return cmake
+
+    def configure(self):
+        version = Version(self.settings.compiler.version)
```
Comment (uilianries) [inline recipes/absent/all/conanfile.py]:
```suggestion
```

File: recipes/absent/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        compiler = self.settings.compiler
```
Comment (uilianries) [inline recipes/absent/all/conanfile.py]:
```suggestion
```

**Comment by rvarago** [comment]
> Remove unused variables

@uilianries I have applied the suggested changes. Thanks for the catch :)

**Pull request merged**